labrad/types.py

```diff
@@ -1319,8 +1337,8 @@ def __flatten_array__(self, a, endianness):
         else:
             elems = (self.elem.flatten(i, endianness=endianness).bytes
                      for i in a.flat)
-            return dims + ''.join(elems), self
-        return dims + a.tostring(), self
+            return dims + b''.join(elems), self
+        return dims + a.tobytes(), self
```

Comment:
`a` is a numpy array, not a string, and numpy arrays have a [`tobytes` method](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.tobytes.html#numpy.ndarray.tobytes) since version 1.9.0 (we're requiring numpy 1.9.1 or later, so this should be ok).